exercises/exercise5/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+)
+
+type Node struct {
+	Value int
+	Next  *Node
+}
+
+func Prepend(head **Node, x int) {
+	// Plan:
+	// * Create new head
+	// * Make current head new head's next
+	// * Addign new head to current head pointer
+
+	var newHead *Node = new(Node)
+	newHead.Value = x
+	newHead.Next = *head
+
+	*head = newHead
+}
+
+func Append(head **Node, x int) {
+
+	var current *Node = *head
+	for current.Next != nil {
+		current = current.Next
+	}
+
+	var appended *Node = new(Node)
```

Comment:
Have you considered using `Prepend` here?

Reply:
Sorry, I can't see exactly in which part you want me to consider its usage

Reply:
Have no you noticed this is exactly the same code as `Prepend`? Except for `appended.Next = nil`, but next node is also `nil`, so it's the same. And double-pointers allow you to have some fun:
```go
func Prepend(n **Node, v int) { *x = &Node{Value: v, Next: *n} }
func Append(n **Node, v int) {
  for *n != nil { n = &(*n).Next }
  Prepend(n)
}
```
Enjoy (-;